### PR TITLE
Using a configurable properties file

### PR DESCRIPTION
**Pull request by seanmcilvenna**
Using a configurable properties file for settings that might change between various environments.

### CODE REVIEW

File: src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java [hunk 1]
```diff
@@ -62,41 +64,47 @@ public DatabaseBackedPagingProvider databaseBackedPagingProvider() {
...
 	@Override
 	@Bean()
 	public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
 		LocalContainerEntityManagerFactoryBean retVal = super.entityManagerFactory();
-		retVal.setPersistenceUnitName("HAPI_PU");
-		retVal.setDataSource(dataSource());
+		retVal.setPersistenceUnitName(HapiProperties.getPersistenceUnitName());
+
+		try {
+			retVal.setDataSource(dataSource());
```
Comment (jamesagnew) [inline src/main/java/ca/uhn/fhir/jpa/demo/FhirServerConfig.java]: It'd probably make more sense to just re-throw the exceptions below, given that we won't be able to start without this succeeding

File: src/main/java/ca/uhn/fhir/jpa/demo/HapiProperties.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            // Load the configurable properties file
+            InputStream in = null;
+
+            try {
+                in = HapiProperties.class.getClassLoader().getResourceAsStream(HAPI_PROPERTIES);
+                HapiProperties.properties = new Properties();
+                HapiProperties.properties.load(in);
+                in.close();
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
```
Comment (jamesagnew) [inline src/main/java/ca/uhn/fhir/jpa/demo/HapiProperties.java]: Same deal here, I'd just re-throw these.

Comment (seanmcilvenna) [reply src/main/java/ca/uhn/fhir/jpa/demo/HapiProperties.java]: I've updated to have handled exceptions re-throw the exceptions as ConfigurationExceptions.

**Comment by seanmcilvenna** [comment]
I don't do a lot of pull requests... I think I addressed your concerns.
Is there something else I need to do??

**Pull request merged**